docs/vm-ee-guide.md

```diff
@@ -0,0 +1,267 @@
+# Onboard
+
+## Table of Contents
+
+<!-- toc -->
+- [VirtualMachines](#virtualmachines)
+    - [Labels](#labels)
+    - [Cloud Tags](#cloud-tags)
+    - [Sample VirtualMachine Output](#sample-virtualmachine-output)
+        - [AWS](#aws)
+        - [Azure](#azure)
+- [ExternalEntity](#externalentity)
+    - [Sample ExternalEntity Output](#sample-externalentity-output)
+        - [Azure](#azure-1)
+    - [Labels](#labels-1)
+    - [Cloud Tags](#cloud-tags-1)
+        - [Usage](#usage)
+<!-- /toc -->
+
+## VirtualMachines
+
+VirtualMachines are onboarded by specifying the matching criteria in `CloudEntitySelector` CR. The virtual machines from
+cloud are imported as `VirtualMachine` objects in `CloudEntitySelector` namespace. `VirtualMachine` object `Name` is a
+unique resource name within the kubernetes cluster. In Azure, ascii sum of the characters in VM Resource ID added as
+suffix to name of Virtual Machine and in AWS, `instance ID` of Virtual Machine is used as `Name`.
+
+### Labels
+
+Meaning of some label fields in `VirtualMachine` objects differ based on the cloud provider type.
+
+| Label        | Description                               | Azure                                    | AWS                          |
+|--------------|-------------------------------------------|------------------------------------------|------------------------------|
+| cloud-vm-uid | Unique identifier of VM                   | `vmId` field on cloud                    | `Instance ID` of VM on cloud |
+| cloud-vpc-id | Unique identifier of VPC                  | `resourceGuid` field on cloud            | `VPC ID` of VPC in cloud     |
+| vpc-name     | VPC object Name within Kubernetes cluster | vpc `Name` and hash of vpc resource `id` | `VPC ID` of VPC in cloud     |
+
+Following labels are generic and related to user configuration.
+
+1. `cpa-name`: Name of `CloudProviderAccount` CR to which the imported vm belongs to.
+2. `cpa-namespace`: Namespace of `CloudProviderAccount` CR to which the imported vm belongs to.
+
+### Cloud Tags
+
+`VirtualMachine` object tags are converted to labels in `ExternalEntity` CR, hence only the tags which comply to
+Kubernetes labels format are imported as tags in vm object.
+
+*Key field specifications*:
+
+- Required field
+- Must be 63 characters or fewer(4 characters are allocated to the prefix "tag-" in labels, hence only 59 characters
```

Comment:
space

Reply:
"Labels are key/value pairs. Valid label keys have two segments: an optional prefix and name, separated by a slash (/). The name segment is required and must be 63 characters or less"

nephe.io comes under prefix. Hence, its not counted towards 63 character limit.